### PR TITLE
feat: pgRouting engine + NAServer route/service-area compat (#1266)

### LINKED ISSUE #1266: Routing engine + NAServer route/service-area compat
> Canonical source of truth: https://github.com/honua-io/agent-delivery-spec/tree/trunk/.specifica/routing-engine-and-naserver-compat/
>
> This issue body is a projection from Specifica. Update the markdown in git, then resync.

## Spec

# Routing engine + NAServer route/service-area compat

## Summary
Stand up a routing engine and expose NAServer route + service-area solves so existing routing clients repoint.

## Why
Stand up a routing engine and expose NAServer route + service-area solves so existing routing clients repoint.

## Scope In
- Select/integrate a routing engine; expose NAServer route + serviceAreas solves.

## Scope Out
- OD matrix + location-allocation (later children).
- Network dataset editing.

## Deliverables
- Routing engine integration + NAServer route/service-area endpoints.

## Acceptance Criteria
- Route and service-area solves return Esri-shaped results for sample networks.

## Canonical References
- https://github.com/honua-io/honua-server/issues/1266

## Coordination
- Child of epic `epic-network-analyst-routing` (honua-io/honua-server).

## Compliance Constraints
- Use open routing data (e.g. OSM) or the customer's own network data. Do not lift Esri network datasets or Esri-licensed street data.
- IP/licensing guardrails coordinated by honua-io/honua-esri-assess#33 (epic-esri-ip-compliance-guardrails). APIs are fair use (Google v. Oracle); this constrains formats, licensed data, and proprietary assets.

## Tasks

# Routing engine + NAServer route/service-area compat

## Tasks
- [ ] Inspect existing conventions in the target repo before implementing.
- [ ] Implement the smallest slice that satisfies the acceptance criteria.
- [ ] Add or update focused tests/smoke evidence.
- [ ] Update docs or release evidence only where this ticket changes behavior.
- [ ] Confirm deferred scope did not leak into the PR.

### PR DESCRIPTION
**Pull request by mikemcdougall**
# Pull Request

## Issue Link
Fixes #1266 (under epic #1265)

## Summary
Stands up a routing capability and exposes Esri **NAServer** `Route/solve` + `ServiceArea/solveServiceArea` solves backed by **open data (pgRouting over PostGIS)**, so existing Esri routing clients can repoint without changing their request shape. Engine choice and the GPL-over-SQL / open-data posture are recorded in **ADR-0050** (pgRouting now; Valhalla documented as the permissive-license fallback). Routing is implemented as a thin protocol adapter over a new, engine-swappable `IRoutingProvider` pipeline — mirroring the existing geocoding module — and the geocoding module was brought to the same quality bar in the same change.

## Changes Made
- **`Honua.Routing` module** — protocol-neutral `IRoutingProvider` + canonical models, `MockRoutingProvider`, and a `PgRoutingProvider` (`pgr_dijkstra` routes with ordered/oriented geometry merge and request-SRID handling; `pgr_drivingDistance` + concave-hull service areas honoring `travelDirection`), plus `RoutingConfiguration` + validator and a provider `Capabilities` surface the adapter gates on.
- **NAServer adapter** — real `Route/solve` and `ServiceArea/solveServiceArea` handlers (Esri param translation → canonical request → provider → Esri JSON), input caps, `f=pjson`, Esri ring-winding normalization. `ClosestFacility` left as the existing stub (out of scope).
- **Migration `043`** — provisions the `ways`/`ways_vertices_pgr` topology, guarded to no-op where pgRouting is absent; dev `docker-compose.yml` + Aspire AppHost use `pgrouting/pgrouting`. CITE/scale/CI images intentionally unchanged.
- **`docker/routing/`** — dedicated pgRouting compose with a deterministic lattice seed and a real OSM ingestion path (`import-osm.sh`, mapconfig, synthetic sample, digest-pinned `osm-import` profile).
- **Telemetry** — `Honua.Routing` and `Honua.Geocoding` ActivitySources with per-solve/per-geocode spans, registered in the shared OTel source list.
- **Geocoding hardening** — namespace aligned to the `Honua.Geocoding` assembly, base provider made internal, CS1591 enforced, ordinal provider-name dispatch.
- **Tests** — NAServer adapter unit + integration (mock), gated `PgRoutingProvider` and end-to-end NAServer-HTTP integration tests over real pgRouting, plus edge cases (non-4326 SRID, unreachable pair, degenerate service area, capability gating).

## Testing
- [x] Unit tests added/updated
- [x] Integration tests added/updated
- [x] Architecture tests pass
- [x] Manual testing performed

## Gate Impact
- [x] PR gates (build, test, governance)
- [ ] Nightly gates (conformance, performance, security)
- [ ] Release gates (packaging, publishing)
- [ ] Deploy gates (promotion, post-apply validation)
- [ ] None — no gate impact

## Docs or Contract Impact
- [ ] OpenAPI spec changed
- [ ] Protobuf/gRPC contract changed
- [ ] Control plane SDK surface changed
- [x] Documentation updated
- [ ] None — no docs or contract impact

## Release/Deploy Impact
- [ ] Requires coordinated release across repos
- [x] Requires database migration
- [x] Requires infrastructure changes
- [ ] Requires environment variable or secret changes
- [ ] None — standard merge-and-release flow

Notes: migration `043` provisions the routing topology (guarded; safe on non-pgRouting images). Dev `docker-compose.yml` and the Aspire AppHost move to the `pgrouting/pgrouting` image; CITE/scale/CI images are unchanged.

## Breaking Changes
None. The `Honua.Geocoding` namespace rename (`Honua.Core.Features.Geocoding.*` → `Honua.Geocoding.Features.Geocoding.*`) is internal to the server and touches no public/SDK contract.

---

## Pre-PR Checklist
- [x] Ran `scripts/ci/pre-pr-check.sh` and all checks passed
- [x] Commit messages follow conventional format: `type: description (#issue)`
- [x] PR title matches main commit message
- [x] Issue number linked above
- [x] Tests added for new functionality
- [ ] If protocol/auth behavior changed: updated compatibility contract
- [ ] If breaking admin/control-plane API changes: updated migration guide
- [ ] If breaking gRPC/proto wire changes: confirmed with explicit review

### CODE REVIEW

File: src/Honua.Routing/Features/Routing/Providers/PgRoutingProvider.cs [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        CancellationToken cancellationToken)
+    {
+        // Merge the visited edge geometries into a single LineString and compute a
+        // geodesic length in meters (ST_Length over geography). ANY(@edge_ids)
+        // keeps the edge id list a single bound array parameter.
+        const string sql = """
+            WITH route_edges AS (
+                SELECT the_geom
+                FROM public.ways
+                WHERE gid = ANY(@edge_ids)
```
[the remark this answers is not in the document]
Comment (mikemcdougall) [reply src/Honua.Routing/Features/Routing/Providers/PgRoutingProvider.cs]: Already addressed in this PR. `MergeEdgeGeometryAsync` no longer uses `gid = ANY(@edge_ids)`; it now joins the ordered `unnest(@edge_ids) WITH ORDINALITY` against `ways`, orients each edge to its traversal node, and builds the geometry with `ST_MakeLine(the_geom ORDER BY ord)`, summing per-traversal length. Repeated/backtracking edges are preserved in order and length is no longer under-reported. Covered by the gated pgRouting integration tests.

File: src/Honua.Routing/Features/Routing/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        var providerName = configuration[ProviderConfigurationKey];
+
+        if (string.Equals(providerName, MockRoutingProvider.ProviderName, StringComparison.OrdinalIgnoreCase))
+        {
+            services.AddScoped<IRoutingProvider, MockRoutingProvider>();
+        }
+        else
+        {
+            // Default: pgRouting-backed provider over the shared connection substrate.
+            services.AddScoped<IRoutingProvider, PgRoutingProvider>();
```
[the remark this answers is not in the document]
Comment (mikemcdougall) [reply src/Honua.Routing/Features/Routing/ServiceCollectionExtensions.cs]: Fixed in a28e0a67. When `Routing:Provider` is unset, pgRouting is now the default only when `DataSource:Provider` is PostgreSQL (or unset). On non-Postgres data providers (DuckDB/MySQL/etc.) an `UnavailableRoutingProvider` is registered instead; it advertises no capabilities, so the NAServer adapter's capability gate returns a clear "not supported by the configured routing provider" error rather than letting pgr_dijkstra SQL fail with a 500. Explicit `Routing:Provider=pgrouting` still opts in regardless of data provider, and `Routing:Provider=none` selects the no-op provider. Added DB-free selection unit tests.
